### PR TITLE
Profiles: ENS Search Input

### PR DESCRIPTION
**Pull request by jxom**
Fixes RNBW-2283

## What changed (plus any additional context for devs)

This PR introduces the `SearchInput` component to be used on the ENS registration flow.

## PoW (screenshots / screen recordings)

https://www.loom.com/share/4ad46b79929241caa5fa57538fe35236

## Dev checklist for QA: what to test

Head to the ENS registration flow and test the following states:
- When the input is empty, it should show a rainbow gradient input,
- When the input is a already registered ENS name, it should show a warning state (yellow gradient),
- When the input is not a registered ENS name, it should show a success state (green gradient)
- When the input is searching, it should show a rainbow gradient and replace the search icon with a spinner

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?

### CODE REVIEW

**Review by estebanmino — COMMENTED** [review]
looking great!! seems like on android the input is different tho, underlined and a bit misaligned for me 
image

also sorry for merging your base branch back in, I couldn't build the app before 😅

File: src/components/ens-registration/SearchInput/SearchInputGradientBackground.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+          width={width}
+        />
+      ))}
+    </>
+  );
+};
+
+export default SearchInputGradientBackground;
+
+//////////////////////////////////////////////////////////////////
```
Comment (estebanmino) [inline src/components/ens-registration/SearchInput/SearchInputGradientBackground.tsx:88]: was this line on purpose?

Comment (jxom) [reply src/components/ens-registration/SearchInput/SearchInputGradientBackground.tsx:88]: Yeah it was – was just visually separating the `AnimatedRadialGradient` from the default component. I'll remove!

**Comment by jxom** [comment]
> looking great!! seems like on android the input is different tho, underlined and a bit misaligned for me image

Yeah, text input seems whack on android. Had to add `keyboardType="password-visible"` to remove the underline.

**Review by estebanmino — APPROVED** [review]

**Pull request merged**